ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -842,300 +1019,362 @@ CMitsubaLoader::SContext::tex_ass_type CMitsubaLoader::getTexture(SContext& ctx,
+		for (uint32_t i = 0u; i < DS1_METADATA_ENTRY_CNT; ++i)
+		{
+			auto& semantic = (*inputs)[i];
+			semantic.type = types[i];
+			semantic.descriptorSection.type = IPipelineMetadata::ShaderInput::ET_UNIFORM_BUFFER;
+			semantic.descriptorSection.uniformBufferObject.binding = ds1layout->getBindings().begin()[0].binding;
+			semantic.descriptorSection.uniformBufferObject.set = 1u;
+			semantic.descriptorSection.uniformBufferObject.relByteoffset = relOffsets[i];
+			semantic.descriptorSection.uniformBufferObject.bytesize = sizes[i];
+			semantic.descriptorSection.shaderAccessFlags = ICPUSpecializedShader::ESS_VERTEX;
+		}
 	}
+
+	return core::make_smart_refctd_ptr<CMitsubaPipelineMetadata>(std::move(_ds0), std::move(inputs));
 }
 
+SContext::SContext(
+	const asset::IGeometryCreator* _geomCreator,
+	const asset::IMeshManipulator* _manipulator,
+	const asset::IAssetLoader::SAssetLoadParams& _params,
+	asset::IAssetLoader::IAssetLoaderOverride* _override,
+	CGlobalMitsubaMetadata* _metadata
+) : creator(_geomCreator), manipulator(_manipulator), params(_params), override_(_override), globalMeta(_metadata),
+	ir(core::make_smart_refctd_ptr<asset::material_compiler::IR>()), frontend(this)
+{
+	//TODO (maybe) dynamically decide which of those are needed OR just wait until IVirtualTexture does it on itself (dynamically creates resident storages)
+	constexpr asset::E_FORMAT formats[]{ asset::EF_R8_UNORM, asset::EF_R8G8_UNORM, asset::EF_R8G8B8_SRGB, asset::EF_R8G8B8A8_SRGB
+#ifdef DERIV_MAP_FLOAT32
+		, asset::EF_R32G32_SFLOAT
+#endif
+	};
+	constexpr size_t formatCount = sizeof(formats)/sizeof(*formats);
+	std::array<asset::ICPUVirtualTexture::ICPUVTResidentStorage::SCreationParams, formatCount> storage;
+	storage[0].formatClass = asset::EFC_8_BIT;
+	storage[0].layerCount = VT_PHYSICAL_PAGE_TEX_LAYERS;
+	storage[0].tilesPerDim_log2 = VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2;
+	storage[0].formatCount = 1u;
+	storage[0].formats = formats;
+
+	storage[1].formatClass = asset::EFC_16_BIT;
+	storage[1].layerCount = VT_PHYSICAL_PAGE_TEX_LAYERS;
+	storage[1].tilesPerDim_log2 = VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2;
+	storage[1].formatCount = 1u;
+	storage[1].formats = formats+1;
+
+	storage[2].formatClass = asset::EFC_24_BIT;
+	storage[2].layerCount = VT_PHYSICAL_PAGE_TEX_LAYERS;
+	storage[2].tilesPerDim_log2 = VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2;
+	storage[2].formatCount = 1u;
+	storage[2].formats = formats+2;
+
+	storage[3].formatClass = asset::EFC_32_BIT;
+	storage[3].layerCount = VT_PHYSICAL_PAGE_TEX_LAYERS;
+	storage[3].tilesPerDim_log2 = VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2;
+	storage[3].formatCount = 1u;
+	storage[3].formats = formats+3;
+
+#ifdef DERIV_MAP_FLOAT32
+	storage[4].formatClass = asset::EFC_64_BIT;
+	storage[4].layerCount = 2u;
+	storage[4].tilesPerDim_log2 = VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2;
+	storage[4].formatCount = 1u;
+	storage[4].formats = formats+4;
+#endif
```

Comment:
deriv map should never be 32bit float XD (16bit SNORM is max)

Reply:
did it for debug purposes when derivative kernel didnt work

Reply:
you can throw it out now, I guess?